ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -245,3 +277,40 @@ message ListUserRightsResponse {
     // All rights of the user.
     repeated Right rights = 1;
 }
+
+
+message AddUserPropertiesRequest {
+    // Required
+    string user_id = 1;
+    // The properties to add.
+    // Ignores adding duplicate properties.
+    // If you need to update a property, you need to delete it first using a separate call.
+    // Required
+    repeated Property properties = 2;
+}
+message AddUserPropertiesResponse {
+    // The properties that were newly added by the request.
+    // Required
+    repeated Property newly_added_properties = 2;
+}
+
+message DeleteUserPropertiesRequest {
+    // Required
+    string user_id = 1;
+    // Properties to delete.
+    // Ignores deleting non-existing properties.
+    // Required
+    repeated Property properties = 2;
+}
+message DeleteUserPropertiesResponse {
+    // The properties that were actually deleted by the request.
+    repeated Property newly_deleted_properties = 1;
+}
+
+message Property {
+    // Required
+    string name = 1;
+    // Optional
+    string value = 2;
```

Comment:
Explicit length restrictions and/or character restrictions should be placed on these fields, like they are with most other strings over the Ledger API.

Reply:
Added restrictions on `name` and made `value` into bytes.